crates/starknet-devnet/tests/test_simulate_transactions.rs

```diff
@@ -232,6 +234,204 @@ mod simulation_tests {
+            .await
+            .unwrap()[0];
+        assert_eq!(resp_no_flags["fee_estimation"]["gas_price"], "0x174876e800");
+        assert_eq!(resp_no_flags["fee_estimation"]["data_gas_price"], "0x174876e800");
+        assert_eq!(resp_no_flags["fee_estimation"]["overall_fee"], "0x7352ecf52000");
+
+        let params_skip_validation = get_params(&["SKIP_VALIDATE"]);
+        let resp_skip_validation = &devnet
+            .send_custom_rpc("starknet_simulateTransactions", params_skip_validation.clone())
+            .await
+            .unwrap()[0];
+        assert_eq!(resp_skip_validation["fee_estimation"]["gas_price"], "0x174876e800");
+        assert_eq!(resp_skip_validation["fee_estimation"]["data_gas_price"], "0x174876e800");
+        assert_eq!(resp_skip_validation["fee_estimation"]["overall_fee"], "0x733ba47e3800");
+
+        assert_difference_if_validation(
+            resp_no_flags,
+            resp_skip_validation,
+            &sender_address_hex,
+            max_fee == FieldElement::ZERO,
+        );
+
+        let gas_update = json!({
+            "gas_price_wei": 9 * 1e18 as u128,
+            "data_gas_price_wei": 8 * 1e18 as u128,
+            "gas_price_strk": 7 * 1e18 as u128,
+            "data_gas_price_strk": 6 * 1e18 as u128,
+            "generate_block": false,
+        });
+        let updated_gas =
+            &devnet.send_custom_rpc("devnet_updateGas", gas_update.clone()).await.unwrap();
+        assert_eq!(updated_gas, &gas_update);
+
+        let chain_id = &devnet.send_custom_rpc("starknet_chainId", json!({})).await.unwrap();
+        assert_eq!(chain_id, assert_chain_id);
+
+        let resp_no_flags = &devnet
+            .send_custom_rpc("starknet_simulateTransactions", params_no_flags)
+            .await
+            .unwrap()[0];
+        assert_eq!(resp_no_flags["fee_estimation"]["gas_price"], "0x7ce66c50e2840000");
+        assert_eq!(resp_no_flags["fee_estimation"]["data_gas_price"], "0x6f05b59d3b200000");
+        assert_eq!(resp_no_flags["fee_estimation"]["overall_fee"], "0x2603cc779e46ad00000");
+
+        let resp_skip_validation = &devnet
+            .send_custom_rpc("starknet_simulateTransactions", params_skip_validation)
+            .await
+            .unwrap()[0];
+        assert_eq!(resp_skip_validation["fee_estimation"]["gas_price"], "0x7ce66c50e2840000");
+        assert_eq!(resp_skip_validation["fee_estimation"]["data_gas_price"], "0x6f05b59d3b200000");
+        assert_eq!(resp_skip_validation["fee_estimation"]["overall_fee"], "0x25fbfe10d93884c0000");
+
+        assert_difference_if_validation(
+            resp_no_flags,
+            resp_skip_validation,
+            &sender_address_hex,
+            max_fee == FieldElement::ZERO,
+        );
+    }
+
+    #[tokio::test]
+    async fn update_gas() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+
+        update_gas_scenario(devnet, "0x534e5f5345504f4c4941").await;
```

Comment:
What's this magical hex string?

Reply:
hex value for mainnet but I'll add comments

Reply:
181e2a3bd263c3cff70f6510836fc750bbc76b2a

Reply:
> Isn't `TESTNET` the default chain ID?

Yes my bad, fixed in ca20adf2cdd9d83e52162771d7b5e4f8969b8f5d